### PR TITLE
changing cmake to require packages since the tests run anyway

### PR DESCRIPTION
**Pull request by PatNichols**
## Description

Some "optional" packages are not treated as such during ctests and build stages. This PR changes the top level CMakeLists.txt to make them mandatory.

## Issue(s) addressed

Resolves  https://github.com/JCSDA-internal/ioda-converters/issues/1253

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -61,20 +62,18 @@ else()
...
-else()
-    message(STATUS "Skipping python tests that import eccodes")
-    set( iodaconv_eccodes_ENABLED False)
+endif()
+
+find_package( cartopy QUIET )
+
+if ( cartopy_FOUND )
+        message(STATUS "cartopy found")
+        set( iodaconv_cartopy_ENABLED True)
```
Comment (fabiolrdiniz) [inline CMakeLists.txt:76]:
```suggestion
        set( iodaconv_cartopy_ENABLED True)
```
```suggestion
        set(iodaconv_cartopy_ENABLED True)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -61,20 +62,18 @@ else()
...
-option(USE_ECCODES "Use Python eccodes package" OFF)
-if ( USE_ECCODES )
-    find_package( eccodes QUIET )
-    if ( eccodes_FOUND )
-        message(STATUS "Adding python tests that import eccodes")
+find_package( eccodes QUIET)
+
+if ( eccodes_FOUND )
+        message(STATUS "eccodes found")
         set( iodaconv_eccodes_ENABLED True)
```
Comment (fabiolrdiniz) [inline CMakeLists.txt:69]:
```suggestion
        set(iodaconv_eccodes_ENABLED True)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -61,20 +62,18 @@ else()
...
-# Optional:  eccodes for Python (and C libs)
-option(USE_ECCODES "Use Python eccodes package" OFF)
-if ( USE_ECCODES )
-    find_package( eccodes QUIET )
-    if ( eccodes_FOUND )
-        message(STATUS "Adding python tests that import eccodes")
+find_package( eccodes QUIET)
+
+if ( eccodes_FOUND )
+        message(STATUS "eccodes found")
```
Comment (fabiolrdiniz) [inline CMakeLists.txt:68]:
```suggestion
        message(STATUS "Found: eccodes")
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -61,20 +62,18 @@ else()
...
-    endif()
-else()
-    message(STATUS "Skipping python tests that import eccodes")
-    set( iodaconv_eccodes_ENABLED False)
+endif()
+
+find_package( cartopy QUIET )
+
+if ( cartopy_FOUND )
+        message(STATUS "cartopy found")
```
Comment (fabiolrdiniz) [inline CMakeLists.txt:75]:
```suggestion
        message(STATUS "Found: cartopy")
```

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @PatNichols!

File: CMakeLists.txt (hunk 1 above)
Comment (PatNichols) [reply CMakeLists.txt:76]: I should probably delete these enviromental variables just in case someone tries to use them later. Thanks for the catch!

**Review by BenjaminRuston — APPROVED** [review]
great let us move forward with this

**Pull request merged**